mr_project/mr_app/views.py

```diff
@@ -1,9 +1,231 @@
+
+    one_emotion_dict = {
+        "emotion": emotion_name,
+        "timestampList": timestamp_list
+    }
+    return one_emotion_dict
+
+
+def fetch_all_emotions_of_one_user(driver_arg, user_name):
+    with driver_arg.session() as session:
+        result = session.run('''MATCH (u)-->(em) WHERE u.name = $user_name  RETURN 
+                             em.name AS emotions''', user_name=user_name)
+        emotion_list = []
+        for record in result:
+            if not record["emotions"] in emotion_list:
+                emotion_list.append(record["emotions"])
+    return emotion_list
+
+
+def fetch_all_emotions_and_epochs_of_user(driver_arg, user_name):
+
+    emotions_of_user = fetch_all_emotions_of_one_user(driver_arg, user_name)
+    list_of_individual_emotion_dicts = []
+
+    for i in range(len(emotions_of_user)):
+        list_of_individual_emotion_dicts.append(fetch_epochs_of_user_emotion_pair(driver_arg, user_name, emotions_of_user[i]))
+
+    dict_of_all_emotions_and_epochs = {
+        "user": user_name,
+        "emotionList": list_of_individual_emotion_dicts
+    }
+
+    return dict_of_all_emotions_and_epochs
+
+
+#FUNCTIONS THAT ACTUALLY CONNECT WITH THE DATABASE TO PERFORM TRANSACTIONS
+def spawn_emotion_node(driver_arg, emotion):
+    with driver_arg.session() as session:
+        emotion_label = session.write_transaction(create_emotion_tx, emotion)
+        print(emotion_label)
+
+
+def spawn_user_node(driver_arg, username):
+    with driver_arg.session() as session:
+        user_name = session.write_transaction(create_user_tx, username)
+        print(user_name)
+
+
+def register_emotion(driver_arg, username, emotion):
+    epoch_timestamp = time()
+    with driver_arg.session() as session:
+        seconds = session.write_transaction(create_epoch_relationship_between_user_and_emotion, emotion, username, epoch_timestamp)
+        print(seconds)
+
+
+def spawn_all_emotions(driver_arg, emotions_list):
+    n_emotions = len(emotions_list)
+    for i in range(n_emotions):
+        spawn_emotion_node(driver_arg, emotions_list[i])
+
+
+def add_users(driver_arg, users_list):
+    n_users = len(users_list)
+    for i in range(n_users):
+        spawn_user_node(driver_arg, users_list[i])
+
+
+@csrf_exempt
+def get_user_emotion(request):
+    if request.method == 'GET':
+        query_arguments = request.GET
+        user = ''
+        emotion = ''
+
+        try:
+            user = query_arguments['user']
+            emotion = query_arguments['emotion']
+        except KeyError:
+            return HttpResponseBadRequest('Missing required key')
+
+        if check_for_user(DRIVER, user) and check_for_emotion(DRIVER, emotion):
+            epoch_dict = fetch_epochs_of_user_emotion_pair(DRIVER, user, emotion)
+
+            return JsonResponse(epoch_dict)
+        return HttpResponseBadRequest('Slow down dawg! The user and/or emotion do not exist, ya need a spelling lesson?')
```

Comment:
It would be helpful if we could add functionality to tell the user exactly what went wrong instead of "one of these things is wrong."

Reply:
yep

Reply:
check... but messy implementation